packages/cozy-sharing/src/DoctypeContact.js

```diff
@@ -1,27 +1,28 @@
+import { models } from 'cozy-client'
 import { Contact as DoctypeContact } from 'cozy-doctypes'
+const ContactModel = models.contact
 
-class Contact extends DoctypeContact {
-  static getInitials(contactOrRecipient, defaultValue = '') {
-    if (Contact.isContact(contactOrRecipient)) {
-      return DoctypeContact.getInitials(contactOrRecipient)
-    } else {
-      const s =
-        contactOrRecipient.public_name ||
-        contactOrRecipient.name ||
-        contactOrRecipient.email
-      return (s && s[0].toUpperCase()) || defaultValue
-    }
+const isContact = candidate => {
+  return candidate._type === 'io.cozy.contacts'
+}
+export const getInitials = (contactOrRecipient, defaultValue = '') => {
+  if (isContact(contactOrRecipient)) {
+    return ContactModel.getInitials(contactOrRecipient)
+  } else {
+    const s =
+      contactOrRecipient.public_name ||
+      contactOrRecipient.name ||
+      contactOrRecipient.email
+    return (s && s[0].toUpperCase()) || defaultValue
   }
+}
 
-  static getDisplayName(contact, defaultValue = '') {
-    if (Contact.isContact(contact)) {
-      return DoctypeContact.getDisplayName(contact)
-    } else {
-      return (
-        contact.public_name || contact.name || contact.email || defaultValue
-      )
-    }
+export const getDisplayName = (contact, defaultValue = '') => {
+  if (isContact(contact)) {
+    return ContactModel.getDisplayName(contact)
+  } else {
+    return contact.public_name || contact.name || contact.email || defaultValue
```

Comment:
Should the two else branch be in a Recipient model ? Would it make sense ?

Reply:
I don't know, I added a todo for the next time